.github/workflows/wasmcloud.yml

```diff
@@ -303,13 +307,14 @@ jobs:
         - provider-archive
         - wascap
 
-    name: publish ${{ matrix.crate }} to crates.io
     if: startsWith(github.ref, 'refs/tags/') && github.event_name == 'push'
     needs: cargo
     runs-on: ubuntu-22.04
     steps:
     - uses: actions/checkout@v4.1.1
-      if: startsWith(github.ref, 'refs/tags/${{ matrix.crate }}-v')
-    - run: cargo publish --token ${{ secrets.CRATES_PUBLISH_TOKEN }}
+      if: startsWith(github.ref, format('refs/tags/{0}-v', matrix.crate))
```

Comment:
This looks equivalent to what we had before 🤔 

I know I must be missing something, since this step ran when we pushed the v0.79.1 tag, but I can't understand how, because the ref would have been `refs/tags/v0.79.1`, and didn't include a `-` at all

Can you point out what I'm missing?

Reply:
Yes, I think `${{ .. }}` reppacement does not work here directly, we need to use a function

Reply:
Ohh, got it!

Reply:
I just found out about `actionlint`, which gives a helpful explanation for this:
```
.github/workflows/wasmcloud.yml:312:11: if: condition "startsWith(github.ref, 'refs/tags/${{ matrix.crate }}-v')" is always evaluated to true because extra characters are around ${{ }} [if-cond]
    |
312 |       if: startsWith(github.ref, 'refs/tags/${{ matrix.crate }}-v')
```